### PR TITLE
Add support for autocert new TLS-ALPN challenge

### PR DESCRIPTION
**Pull request by frankban**
Rather than using the HTTP challenge, requiring port 80 to be open, and therefore an additional handler, we use the new TLS-ALPN challenge, which works using the TLS handshake on 443.

**QA:**
Assuming $DNSNAME is a DNS name that you control:
- bootstrap juju with `--config autocert-dns-name=$DNSNAME`; 
- when the bootstrap process completes, point $DNSNAME to the ip address of the controller;
- wait for the change to propagate;
- run `juju gui`: the output should suggest a URL to visit which includes $DNSNAME;
- visit the GUI at that URL, and it should work;
- deploying charms, running `juju status`, i.e. accessing the API should work.

### CODE REVIEW

File: apiserver/apiserver.go [hunk 1]
```diff
@@ -290,9 +290,9 @@ func newServer(cfg ServerConfig) (_ *Server, err error) {
 	srv.logSinkWriter = logSinkWriter
 
 	if cfg.PrometheusRegisterer != nil {
-		apiserverCollectior := NewMetricsCollector(&metricAdaptor{srv})
```
Comment (frankban) [inline apiserver/apiserver.go]: Drive-by fix of a typo.

Comment (SimonRichardson) [reply apiserver/apiserver.go]: :+1:

**Comment by frankban** [comment]
Thanks for the review!
$$merge$$
